### PR TITLE
Compose refactor

### PR DESCRIPTION
**Pull request by mixmix**
selection_160

@ahdinosaur @mmckegg this is ready for a review

### CODE REVIEW

File: modules_basic/compose.js [hunk 1]
```diff
@@ -21,59 +22,64 @@ exports.needs = {
...
-    var accessories
-    meta = meta || {}
-    if(!meta.type) throw new Error('message must have type')
-    var ta = h('textarea', {
+    var actions
+
+    var textArea = h('textarea', {
       placeholder: opts.placeholder || 'Write a message',
-      style: {height: opts.shrink === false ? '200px' : ''}
+      className: opts.shrink === false ? '-expanded' : '-contracted'
```
Comment (ahdinosaur) [inline modules_basic/compose.js]: like the class names, but maybe we should move the class to the top "Compose" element?

File: modules_basic/compose.js [hunk 2]
```diff
@@ -111,45 +117,46 @@ exports.create = function (api) {
...
-
-    suggest(ta, function (name, cb) {
+      var embed = file.type.indexOf('image/') === 0 ? '!' : ''
+      textArea.value += '\n' + embed + '['+file.name+']('+file.link+')'
+      console.log('added:', file)
+    })
+    var publishBtn = h('button', {'ev-click': publish}, 'Publish' )
+    var actions = h('section.actions', 
+      //hidden until you focus the textarea
+      { style: {display: opts.shrink === false ? '' : 'none'} },
```
Comment (ahdinosaur) [inline modules_basic/compose.js]: can we do this by re-using the expanded / contracted class?

File: modules_basic/compose.mcss [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      $composeButton
+
+      text-transform: uppercase
+      font-weight: bold
+      font-size: .7rem
+    }
+  }
+}
+
+$composeButton {
```
Comment (ahdinosaur) [inline modules_basic/compose.mcss]: rad

File: modules_basic/compose.js [hunk 4]
```diff
@@ -21,59 +22,64 @@ exports.needs = {
...
-    shrink: boolean. set to false, to make composer not shrink (or hide controls) when unfocused.
-*/
+      placeholder: string. placeholder text, defaults to "Write a message"
+      prepublish: function. called before publishing a message.
+      shrink: boolean. set to false, to make composer not shrink (or hide controls) when unfocused.
+  */
 
-exports.create = function (api) {
+  function message_compose (meta = {}, opts = {}, cb) {
+    if(!meta.type) throw new Error('message must have type')
```
Comment (mixmix) [inline modules_basic/compose.js]: can we do this in a different pull request ?
I reckon rolling that sort of pattern out in scale across the code base could be really good.

File: modules_basic/compose.js (hunk 1 above)
Comment (mixmix) [inline modules_basic/compose.js]: good idea, I'll try that and see how it looks

File: modules_basic/compose.js (hunk 2 above)
Comment (mixmix) [inline modules_basic/compose.js]: good noticing !

Comment (mixmix) [inline modules_basic/compose.js]: done

File: modules_basic/compose.js (hunk 1 above)
Comment (mixmix) [inline modules_basic/compose.js]: done

**Comment by mixmix** [comment]
Thanks for the review @ahdinosaur !
